Enable KEY_PERFORMANCE key present on Linux 6.17

mhalano opened the pull request:
Linux 6.17 defines a key called KEY_PERFORMANCE for machines that have a perfomance mode, like Alienware and Dell G-series.

bluca commented:
What happens with this change when running on previous kernels?

mhalano commented:
@bluca Good question: it just doesn't work. No negative side effects. Since older kernels don't recognize the key yet, it will generate nothing.
I doubt that's a problem, since no application at the user level uses this key.
I will release soon a user-space code to detect the key but I hope in the future tools like power-profiles-daemon become responsible for such detection.

kuu-rt commented:
Hi Marcos,

I suggest you submit the KEY_PERFORMANCE Linux [commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=89c52146392948f4cdda3853da9d82ec6d1dd1f4) for stable inclusion before this goes through. Otherwise users will lose control of this key.

You can find a guide [here](https://docs.kernel.org/process/stable-kernel-rules.html).

mhalano commented:
@kuu-rt Sorry, I didn't understand. You mean doing a backport to 6.16? The commit is already in 6.17-rc1, so soon it will be in 6.17 stable.

yuwata commented:
Hm, or, not familiar, but is it work even with an older kernel but systemd is built with new input-event-codes.h header file?

mhalano commented:
I don't know for sure. It's possible, though. Do you have a suggestion how to test it? I don't know which part of systemd uses the header...

yuwata commented:
systemd has copy of kernel headers in src/include/uapi/linux/, and the kernel key code header is used to build systemd-udevd.
So, please try
1. copy newer headers from kernel v6.17-rc1 or newer there,
2. apply your patch to hwdb,
3. build systemd,
4. then check if it works with kernel 6.16 or order.

mhalano commented:
I updated the header file, added the KEY_PERFORMANCE in the evtest and compiled, so now I get:
```
Event: time 1754903516.136114, -------------- SYN_REPORT ------------
Event: time 1754903516.136149, type 4 (EV_MSC), code 4 (MSC_SCAN), value 68
Event: time 1754903516.136149, type 1 (EV_KEY), code 701 (KEY_PERFORMANCE), value 0
Event: time 1754903516.136149, -------------- SYN_REPORT ------------
```
So I think we are good, right? We just need to use an updated header.

yuwata commented:
OK, then please update this PR as
1. update kernel headers from 6.17-rc1. See README file in our kernel header directory.
2. Apply your change to hwdb.

mhalano commented:
@yuwata One question: do you think these changes can enter v258 or is it too late for that?
